applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp

```diff
@@ -861,58 +904,33 @@ namespace MPMParticleGeneratorUtility
         if (!IsEqualVolumes) rN = rGeom.ShapeFunctionsValues(rIntegrationMethod);
     }
 
-    void DetermineConditionIntegrationMethodAndShapeFunctionValues(const GeometryType& rGeom, const SizeType ParticlesPerCondition,
-        IntegrationMethod& rIntegrationMethod, Matrix& rN, bool& IsEqualVolumes)
+    void DetermineConditionIntegrationMethod(const GeometryType& rGeom, const SizeType ParticlesPerCondition,
+        IndexType& rNumPointsPerSpan)
     {
         const GeometryData::KratosGeometryType geo_type = rGeom.GetGeometryType();
         const SizeType domain_size = rGeom.WorkingSpaceDimension();
 
-        if (geo_type == GeometryData::KratosGeometryType::Kratos_Point2D  || geo_type == GeometryData::KratosGeometryType::Kratos_Point3D)
-        {
-            switch (ParticlesPerCondition)
-            {
-                case 0: // Default case
-                    IsEqualVolumes = true;
-                    rN = ZeroMatrix(1,1);
-                    break;
-                case 1: // Only nodal
-                    IsEqualVolumes = true;
-                    rN = ZeroMatrix(1,1);
-                    break;
-                default:
-                    IsEqualVolumes = true;
-                    rN = ZeroMatrix(1,1);
-                    std::string warning_msg = "The input number of PARTICLES_PER_CONDITION: " + std::to_string(ParticlesPerCondition);
-                    warning_msg += " is not available for Point" + std::to_string(domain_size) + "D.\n";
-                    warning_msg += "Available option is: 1 (default).\n";
-                    warning_msg += "The default number of particle: 1 is currently assumed.";
-                    KRATOS_INFO("MPMParticleGeneratorUtility") << "WARNING: " << warning_msg << std::endl;
-                    break;
-            }
-
-
-
-        }
-        else if (geo_type == GeometryData::KratosGeometryType::Kratos_Line2D2  || geo_type == GeometryData::KratosGeometryType::Kratos_Line3D2)
+        if (geo_type == GeometryData::KratosGeometryType::Kratos_Line2D2  || geo_type == GeometryData::KratosGeometryType::Kratos_Line3D2)
         {
             switch (ParticlesPerCondition)
```

Comment:
Can't you just make:
rNumber.. = (ParticlesPerCondition < 5) ? ParticlesPerCondition : default;

Reply:
you are right I can simplify that - thanks